code/game/objects/items/weapons/cigs_lighters.dm

```diff
@@ -776,4 +774,81 @@ MATCHBOXES ARE ALSO IN FANCY.DM
 		playsound(get_turf(src), pick(close_sound), 50, 1)
 		user.visible_message("<span class='rose'>You hear a quiet click as [user] shuts off \the [src] without even looking at what they're doing. Wow.</span>", \
 		"<span class='rose'>You hear a quiet click as you shut off \the [src] without even looking at what you are doing.</span>")
-	update_brightness()
\ No newline at end of file
+	update_brightness()
+
+////////
+//VAPE//
+////////
+
+/obj/item/weapon/vape
+	name = "\improper E-Cigarette"
+	desc = "More commonly referred to as a mouth fedora"
+	icon = 'icons/obj/items.dmi'
+	icon_state = "vape"
+	var/datum/effect/effect/system/smoke_spread/smoke = new()
+	var/max_beaker_volume = 50
+	var/beaker = null
```

Comment:
Cast this to a beaker always so you won't have to cast it in the attack self code.